snuba/clusters/cluster.py

```diff
@@ -408,9 +443,20 @@ def get_cluster(storage_set_key: StorageSetKey) -> ClickhouseCluster:
     assert (
         storage_set_key not in DEV_STORAGE_SETS or settings.ENABLE_DEV_FEATURES
     ), f"Storage set {storage_set_key} is disabled"
-    res = _get_storage_set_cluster_map().get(storage_set_key, None)
-    if res is None:
-        raise UndefinedClickhouseCluster(
-            f"{storage_set_key} is not a defined in the CLUSTERS setting for this environment"
-        )
+
+    if partition_id:
```

Comment:
If a storage set is partitioned, it would provide a different storage set key with a suffix like `_1` without passing the partition_id. Will that behavior still work with this change?

Reply:
We're not using storage set key alone to route with this approach.

A single storage set can map to multiple clusters based on the (storage set key, partition ID) pair. If we go with this approach, you'll want your mapping system to pass partition_id here, there won't be a re-routing to another storage set key.